lib/screens/create_region_screens/interactor/viewmodels/create_region_bloc.dart

```diff
@@ -28,20 +30,51 @@ class CreateRegionBloc extends Bloc<CreateRegionEvent, CreateRegionState> {
 
   void _onRegionNameChange(OnRegionNameChange event, Emitter<CreateRegionState> emit) {
     if (event.regionName.isEmpty) {
-      emit(state.copyWith(regionName: event.regionName, isRegionNameNextAvailable: false));
+      emit(state.copyWith(regionName: event.regionName, isRegionNameNextButtonEnable: false));
     } else {
       emit(
-        state.copyWith(regionName: event.regionName, isRegionNameNextAvailable: true),
+        state.copyWith(regionName: event.regionName, isRegionNameNextButtonEnable: true),
       );
     }
   }
 
+  void _onRegionNameNextTapped(OnRegionNameNextTapped event, Emitter<CreateRegionState> emit) {
+    if (state.regionId.isEmpty) {
+      String suggestedRegionId = state.regionName.toLowerCase().trim().split('').map((character) {
```

Comment:
wait what? 

whats going on here? 

Can you explain this logic. 

Also leave comments on whats going on here and why

Reply:
none of this logic should be in the bloc. move to a mapper

Reply:
Generate a suggested Region Id for the user based on the Region name they entered

Reply:
I moved logic to a mapper

Reply:
Fixed logic, Id can actually be less than 8.